File: week01/red-sprout/solution02.py
zeros = 0


def binary(num):
    s = ""
    while num > 0:
        s = str(num & 1) + s
        num = num >> 1
    return s


def convert(s):
    global zeros

    l = len(s)
    cnt = 0

    for i in range(l):
        c = s[i]
        if c == '1':
            cnt += 1
        else:
            zeros += 1

    return binary(cnt)


def loop(s):
    cnt = 0

    while s != "1":
        s = convert(s)
        cnt += 1

    return [cnt, zeros]


def solution(s):
    global zeros
    zeros = 0
    return loop(s)

File: week01/red-sprout/solution02.java
import java.util.*;

class Solution {
    int zeros;

    String binary(int num) {
        String s = "";
        while(num > 0) {
            s = (num & 1) + s;
            num = num >> 1;
        }
        return s;
    }

    String convert(String s) {
        int l = s.length();
        int cnt = 0;
        for(int i = 0; i < l; ++i) {
            char c = s.charAt(i);
            if(c == '1') {
                ++cnt;
            } else {
                ++zeros;
            }
        }
        return binary(cnt);
    }

    int[] loop(String s) {
        int cnt = 0;
        while(!s.equals("1")) {
            s = convert(s);
            ++cnt;
        }
        return new int[] {cnt, zeros};
    }

    public int[] solution(String s) {
        zeros = 0;
        return loop(s);
    }
}
